tests/test_iotedgedev.py

```diff
@@ -1,266 +1,385 @@
     assert "Module name cannot start or end with the symbol _" in result.output
 
-    result = runner.invoke(cli.main, ["solution", "add", "csharpmodule_", "--template", "csharp"])
-    print(result.output)
+    result = runner_invoke(["solution", "add", "csharpmodule_", "--template", "csharp"], True)
     assert "Module name cannot start or end with the symbol _" in result.output
 
-    result = runner.invoke(cli.main, ["solution", "add", "csharp-module", "--template", "csharp"])
-    print(result.output)
+    result = runner_invoke(["solution", "add", "csharp-module", "--template", "csharp"], True)
     assert "Module name can only contain alphanumeric characters and the symbol _" in result.output
 
-    result = runner.invoke(cli.main, ["solution", "add", "filtermodule", "--template", "csharp"])
-    print(result.output)
+    result = runner_invoke(["solution", "add", "filtermodule", "--template", "csharp"], True)
     assert "already exists under" in result.output
 
 
 @pytest.fixture
-def test_push_modules(request):
-
-    os.chdir(test_solution_dir)
-
-    cli = __import__("iotedgedev.cli", fromlist=['main'])
-    runner = CliRunner()
-    result = runner.invoke(cli.main, ['push', '-P', get_platform_type()])
-    print(result.output)
-    print(result.exception)
+def test_push_modules():
+    result = runner_invoke(['push', '-P', get_platform_type()])
 
     assert 'BUILD COMPLETE' in result.output
     assert 'PUSH COMPLETE' in result.output
     assert 'ERROR' not in result.output
 
 
 @pytest.fixture
-def test_deploy_modules(request):
-
-    os.chdir(test_solution_dir)
-
-    cli = __import__("iotedgedev.cli", fromlist=['main'])
-    runner = CliRunner()
-
-    result = runner.invoke(cli.main, ['deploy', '-f', os.path.join('config', 'deployment.' + get_platform_type() + '.json')])
-    print(result.output)
+def test_deploy_modules():
+    if get_docker_os_type() == "windows":
+        result = runner_invoke(['deploy', '-f', os.path.join('config', 'deployment.' + get_platform_type() + '.json')])
+    else:
+        result = runner_invoke(['deploy'])
 
     assert 'DEPLOYMENT COMPLETE' in result.output
     assert 'ERROR' not in result.output
 
 
-def test_monitor(request, capfd):
-
-    os.chdir(test_solution_dir)
+@pytest.fixture
+def test_monitor(capfd):
+    runner_invoke(['monitor', '--timeout', '5'])
 
-    cli = __import__("iotedgedev.cli", fromlist=['main'])
-    runner = CliRunner()
-    result = runner.invoke(cli.main, ['monitor', '--timeout', '5'])
     out, err = capfd.readouterr()
-    print(out)
-    print(err)
-    print(result.output)
-    print(result.exception)
-
     if not PY35:
         assert 'Monitoring events from device' in out
     else:
         assert not err
 
 
-def test_e2e(test_push_modules, test_deploy_modules):
-    print('Testing E2E')
+def test_e2e(prepare_solution_with_env, test_push_modules, test_deploy_modules, test_monitor):
```

Comment:
the input of the methods are not called?

Reply:
what you mean? There is no parameter